hugegraph-core/src/main/java/org/apache/hugegraph/masterelection/GlobalMasterInfo.java

```diff
@@ -19,26 +19,20 @@
 
 public class GlobalMasterInfo {
 
-    private boolean isMaster;
-    private String url;
-
+    private volatile Info info;
     private volatile boolean featureSupport;
 
     public GlobalMasterInfo() {
         this.featureSupport = false;
+        this.info = new Info(false, "");
     }
 
-    public synchronized void set(boolean isMaster, String url) {
-        this.isMaster = isMaster;
-        this.url = url;
-    }
-
-    public synchronized boolean isMaster() {
-        return this.isMaster;
+    public void info(boolean isMaster, String url) {
+        this.info = new Info(isMaster, url);
```

Comment:
`volatile Info info = new Info(isMaster, url);` 

Does `volatile` guarantee the atomic integrity of `Info` initialization? Or shall we consider to use `final` to guarantee to complete initialization like:

`final Info info = new Info(isMaster, url);`

Reply:
info define with volatile, so as my think, the read  operator happend-before write operator,
we read the info must is init finished;

Reply:
> info define with volatile, so as my think, the write operator happens-before read operator, so if write operator happend first, the read operator must after the write finish

```java
        public Info(boolean isMaster, String url) {
            this.isMaster = isMaster; // ①
            this.url = url; // ②
        }
```

`new Info` include ① & ② step, when ① is finished but ② is not, will we see an inconsistent assignments?

like `Info info = new Info(true, "a")`, will we get `Info(true, "")` or `Info(false, "a")`?

Reply:
the case don't happend, because method parameter be bound;
when call 1 step, the isMaster and url parameter never be modify by external logic, because the boolean is copy the origin data, and string is final;

this is a interesting question:
``` 
      boolean isMaster = global.isMaster(); // 1
      String url = global.url(); //2
      this.info = new Info(isMaster, url); // 3
```
if the code run in more than one thread, the timing might look like this:
thread1: run 1 step; thread2: modify global.master; thread2: modify global.url; thread1: run 2; this case will have some concurrency problem; 
but in our case, only one thread modify global, so here is no problem

Reply:
here is a “copy on write” thought;
single thread copy the data and modify it, then multi-thread read data don't consider lock